### PR TITLE
typeahead core

### PR DESCRIPTION
**Pull request by ymatsuda**

### CODE REVIEW

**Comment by raymondng** [comment]
awesome :+1:  Thanks @ymatsuda!

File: kifi-backend/modules/shoebox/app/com/keepit/socialtypeahead/PrefixFilter.scala [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  private[socialtypeahead] def genFilter(inputName: String): Long = {
+    val name = normalize(inputName)
+    var filter = 0L
+    val maxPrefixLen = min(numHashFuncs.length, name.length)
+    var i = 0
+    while (i < maxPrefixLen) {
+      val prefix = name.subSequence(0, i + 1)
+      var hash = init(prefix.hashCode.toLong)
+      var j = numHashFuncs(i)
+      while (j > 0) {
```
Comment (yingjieMiao) [inline kifi-backend/modules/shoebox/app/com/keepit/socialtypeahead/PrefixFilter.scala:49]: infinite loop? i, j not varying?

Comment (raymondng) [inline kifi-backend/modules/shoebox/app/com/keepit/socialtypeahead/PrefixFilter.scala:49]: Yeah I added the increments locally -- I think Yasu is checking whether we
are paying attn :)

Comment (ymatsuda) [inline kifi-backend/modules/shoebox/app/com/keepit/socialtypeahead/PrefixFilter.scala:49]: exactly :)

**Pull request merged**